packages/react/src/Popover/Root/usePopoverRoot.ts

```diff
@@ -69,22 +68,21 @@ export function usePopoverRoot(params: usePopoverRoot.Parameters): usePopoverRoo
     (nextOpen: boolean, event?: Event, reason?: OpenChangeReason) => {
       onOpenChange(nextOpen, event, reason);
       setOpenUnwrapped(nextOpen);
-      if (!keepMounted && !nextOpen) {
```

Comment:
Need to check how `setMounted(false)` with this `keepMounted` check operates now for the other components with the `hidden` prop

Reply:
Looks like `keepMounted` was never passed to the hooks, hence it worked like `keepMounted=false`. I've removed the prop for Popover but it should be done for all components.